cmake/modules/config.cmake

```diff
@@ -43,5 +43,9 @@ if (WIN32_USE_STDIO OR APPLE)
 	SET (NOCOLOR Yes)
 ENDIF()
 
+IF (IRIX)
+	SET (DECLARE_STRLCPY 1)
+ENDIF()
```

Comment:
Like this?

Reply:
CHECK_FUNCTION_EXISTS("strlcpy" HAVE_STRLCPY), which we actually already do. So just check against that define.

Reply:
This doesn't work for me (`#ifdef HAVE_STRLCPY`). My cmake says none was found but CMake doc mentions some possible false negative cases: https://cmake.org/cmake/help/v3.8/module/CheckFunctionExists.html

Reply:
Are you saying you have it and it ends up defined?

Reply:
There is `strldup` for me obviously, but Cmake says the contrary.

Reply:
Btw. I'd still need to add the entry to `cmake_config.h.in` in case. Or do something different to this check.

Reply:
I guess you meant strlcpy. It's a BSD function, so you shouldn't have it — it's odd that you do. Are you sure the error doesn't stem from our definition (can be ifdef-ed the same way). In any case, the check needs to be fixed if it's broken or removed.

Reply:
Yes, of course `strlcpy`. I'll check that.

Reply:
Mystery solved: It just didn't cover with `core/System/String.h` previous' signature because of `GEM_EXPORT`.